src/llscan.cc

```diff
@@ -645,11 +650,55 @@ void FindReferencesCmd::PrintReferences(SBCommandReturnObject& result,
       // Basically we need to access objects and arrays as both objects and
       // arrays.
       v8::JSObject js_obj(heap_object);
-      scanner->PrintRefs(result, js_obj, err);
+      scanner->PrintRefs(result, js_obj, err, level);
+
+      if (options->recursive_scan) {
```

Comment:
This if block seems identical to lines 682-699 and 824-843. Can we reuse this somehow?